### DISCUSSION TITLE
How to bootstrap Resteasy Client in EJB container

### DISCUSSION THREAD

**Discussion by xqvier** [resteasy/resteasy#3334] [General]
When using resteasy client in an EJB container (i.e: jboss EAP 7.3 environment - resteasy version: 3.15.3.Final), some features arent working as expected.

To name a few:
- The property resteasy.preferJacksonOverJsonB can only be setted as a system property (or from servlet context param which are unusable outside a war). This isn't ideal in a context where we deploy multiple application so they don't interfere with each other. 
I think the documentation is misleading about that and let us think that this property is settable via microprofile-config. I don't know if I did this correctly but after a quick look at the [code](https://github.com/resteasy/resteasy/blob/3.15.3.Final/providers/json-binding/src/main/java/org/jboss/resteasy/plugins/providers/jsonb/JsonBindingProvider.java#L49), this doesnt seem to be considered.
What would be the correct way to go around that ? The only workaround I found is to exclude the json-binding-provider from my `jboss-deployment-structure.xml`.

- ContextResolver, be it a `ContextResolver<Jsonb>` or a `ContextResolver<ObjectMapper>` aren't picked up. In a servlet context, thoses `@Provider` are picked up by the `ListenerBootstrap.getParameter(ResteasyContextParameters.RESTEASY_SCANNED_PROVIDERS)` in [ConfigurationBootstrap#L132](https://github.com/resteasy/resteasy/blob/3.15.3.Final/resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/server/servlet/ConfigurationBootstrap.java#L132)

Any idea how to solve these issues ? If I understood correctly, the [documentation](https://docs.jboss.org/resteasy/docs/3.15.2.Final/userguide/html/Installation_Configuration.html#config_mechanisms) only tell us how to bootstrap inside a servlet container which is an obvious prerequisite when **serving** REST request. But I think that **calling** REST API from an EJB container isn't a weird usecase.

**Comment by jamezp** [comment]
RESTEasy is only processed for web archives, WAR's. While other deployment types, e.g. EJB JAR's, will get the dependencies added, they won't be processed like a typical Jakarta REST deployment. The [specification](https://jakarta.ee/specifications/restful-ws/3.1/jakarta-restful-ws-spec-3.1.html#servlet) itself specifically mentions what should happen in a servlet container and a WAR. Your `ContextResolver`'s will not be resolved in a JAR deployment. Either way, for a client they are not resolved anyway and you'd need to explicitly register them on the client.

That said as you see the client does work, however you'd need to use the {{jboss-deployment-structure.xml}} to use the client with the specific dependencies you'd like.
